contracts/common/StorageAccessible.sol

```diff
@@ -43,13 +43,18 @@ abstract contract StorageAccessible {
         /* solhint-disable no-inline-assembly */
         /// @solidity memory-safe-assembly
         assembly {
+            let gasBefore := gas()
             let success := delegatecall(gas(), targetContract, add(calldataPayload, 0x20), mload(calldataPayload), 0, 0)
+            let gasAfter := gas()
+            let gasUsed := sub(gasBefore, gasAfter)
```

Comment:
This can be returned by the `targetContract`. You can check the `SimulateTxAccessor` contract as an example. You can basically combine its approach with your original contract for detailed simulations